service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -77,21 +82,122 @@
      */
     public void persistHiddenDirectionValues(List<Element<Direction>> directionWithValues,
                                              List<Element<Direction>> directionToAddValues) {
-        directionToAddValues
-            .forEach(directionToAddValue -> directionWithValues
-                .stream()
-                .filter(direction ->
-                    direction.getValue().getDirectionType().equals(directionToAddValue.getValue().getDirectionType()))
-                .forEach(direction -> {
-                    directionToAddValue.getValue().setReadOnly(direction.getValue().getReadOnly());
-                    directionToAddValue.getValue().setDirectionRemovable(direction.getValue().getDirectionRemovable());
-                    directionToAddValue.getValue().setAssignee(defaultIfNull(
-                        directionToAddValue.getValue().getAssignee(), direction.getValue().getAssignee()));
-
-                    if (!direction.getValue().getReadOnly().equals("No")) {
-                        directionToAddValue.getValue().setDirectionText(direction.getValue().getDirectionText());
-                    }
-                }));
+        directionToAddValues.forEach(directionToAddValue -> directionWithValues
+            .stream()
+            .filter(direction ->
+                direction.getValue().getDirectionType().equals(directionToAddValue.getValue().getDirectionType()))
+            .forEach(direction -> {
+                directionToAddValue.getValue().setReadOnly(direction.getValue().getReadOnly());
+                directionToAddValue.getValue().setDirectionRemovable(direction.getValue().getDirectionRemovable());
+                directionToAddValue.getValue().setAssignee(defaultIfNull(
+                    directionToAddValue.getValue().getAssignee(), direction.getValue().getAssignee()));
+
+                if (!direction.getValue().getReadOnly().equals("No")) {
+                    directionToAddValue.getValue().setDirectionText(direction.getValue().getDirectionText());
+                }
+            }));
+    }
+
+    /**
+     * Adds a response by an assignee to a direction where the direction ID of the response matches the direction ID
+     * of the direction.
+     *
+     * @param responses  a list of single responses to directions.
+     * @param directions a list of directions with combined responses.
+     */
+    public void addResponsesToDirections(List<DirectionResponse> responses, List<Element<Direction>> directions) {
+        directions.forEach(direction -> responses.stream()
```

Comment:
could `directions` and `responses` be null at any point?

Reply:
no